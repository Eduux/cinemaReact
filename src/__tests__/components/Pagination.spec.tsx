/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render, fireEvent, getByText, act } from '@testing-library/react';

import Component from 'components/Pagination';

describe('Pagination component', () => {
  const initialState = {
    page: 0,
    pages: 0,
    onChange: () => {},
  };

  it('should render without data', () => {
    const result = render(<Component {...initialState} />);
    expect(result).toMatchSnapshot();
  });

  it('should render with data', () => {
    const result = render(<Component {...initialState} pages={5} />);
    expect(result).toMatchSnapshot();
  });

  it('should render with data and change page', () => {
    const changePageMock = jest.fn();

    const result: any = render(
      <Component {...initialState} pages={5} onChange={changePageMock} />,
    );
    const { container } = result;

    act(() => {
      fireEvent(
        getByText(container, 'Pr√≥ximo'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );

      fireEvent(
        getByText(container, '1'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );

      fireEvent(
        getByText(container, 'Anterior'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    });

    expect(changePageMock).toHaveBeenCalledWith(0);
    expect(changePageMock).toHaveBeenCalledWith(1);
    expect(result).toMatchSnapshot();
  });
});
