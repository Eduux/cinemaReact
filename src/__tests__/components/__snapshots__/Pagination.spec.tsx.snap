// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvaa iCgHVd"
      >
        <ul>
          <li
            class="previous disabled"
          >
            <a
              aria-disabled="true"
              aria-label="Previous page"
              rel="prev"
              role="button"
              tabindex="0"
            >
              Anterior
            </a>
          </li>
          <li
            class="selected"
          >
            <a
              aria-current="page"
              aria-label="Page 1 is your current page"
              role="button"
              tabindex="0"
            >
              1
            </a>
          </li>
          <li>
            <a
              aria-label="Page 2"
              role="button"
              tabindex="0"
            >
              2
            </a>
          </li>
          <li>
            <a
              aria-label="Page 3"
              role="button"
              tabindex="0"
            >
              3
            </a>
          </li>
          <li>
            <a
              aria-label="Page 4"
              role="button"
              tabindex="0"
            >
              4
            </a>
          </li>
          <li>
            <a
              aria-label="Page 5"
              role="button"
              tabindex="0"
            >
              5
            </a>
          </li>
          <li
            class="next"
          >
            <a
              aria-disabled="false"
              aria-label="Next page"
              rel="next"
              role="button"
              tabindex="0"
            >
              Pr贸ximo
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvaa iCgHVd"
    >
      <ul>
        <li
          class="previous disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            rel="prev"
            role="button"
            tabindex="0"
          >
            Anterior
          </a>
        </li>
        <li
          class="selected"
        >
          <a
            aria-current="page"
            aria-label="Page 1 is your current page"
            role="button"
            tabindex="0"
          >
            1
          </a>
        </li>
        <li>
          <a
            aria-label="Page 2"
            role="button"
            tabindex="0"
          >
            2
          </a>
        </li>
        <li>
          <a
            aria-label="Page 3"
            role="button"
            tabindex="0"
          >
            3
          </a>
        </li>
        <li>
          <a
            aria-label="Page 4"
            role="button"
            tabindex="0"
          >
            4
          </a>
        </li>
        <li>
          <a
            aria-label="Page 5"
            role="button"
            tabindex="0"
          >
            5
          </a>
        </li>
        <li
          class="next"
        >
          <a
            aria-disabled="false"
            aria-label="Next page"
            rel="next"
            role="button"
            tabindex="0"
          >
            Pr贸ximo
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pagination component should render with data and change page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvaa iCgHVd"
      >
        <ul>
          <li
            class="previous"
          >
            <a
              aria-disabled="false"
              aria-label="Previous page"
              rel="prev"
              role="button"
              tabindex="0"
            >
              Anterior
            </a>
          </li>
          <li>
            <a
              aria-label="Page 1"
              role="button"
              tabindex="0"
            >
              1
            </a>
          </li>
          <li
            class="selected"
          >
            <a
              aria-current="page"
              aria-label="Page 2 is your current page"
              role="button"
              tabindex="0"
            >
              2
            </a>
          </li>
          <li>
            <a
              aria-label="Page 3"
              role="button"
              tabindex="0"
            >
              3
            </a>
          </li>
          <li>
            <a
              aria-label="Page 4"
              role="button"
              tabindex="0"
            >
              4
            </a>
          </li>
          <li>
            <a
              aria-label="Page 5"
              role="button"
              tabindex="0"
            >
              5
            </a>
          </li>
          <li
            class="next"
          >
            <a
              aria-disabled="false"
              aria-label="Next page"
              rel="next"
              role="button"
              tabindex="0"
            >
              Pr贸ximo
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvaa iCgHVd"
    >
      <ul>
        <li
          class="previous"
        >
          <a
            aria-disabled="false"
            aria-label="Previous page"
            rel="prev"
            role="button"
            tabindex="0"
          >
            Anterior
          </a>
        </li>
        <li>
          <a
            aria-label="Page 1"
            role="button"
            tabindex="0"
          >
            1
          </a>
        </li>
        <li
          class="selected"
        >
          <a
            aria-current="page"
            aria-label="Page 2 is your current page"
            role="button"
            tabindex="0"
          >
            2
          </a>
        </li>
        <li>
          <a
            aria-label="Page 3"
            role="button"
            tabindex="0"
          >
            3
          </a>
        </li>
        <li>
          <a
            aria-label="Page 4"
            role="button"
            tabindex="0"
          >
            4
          </a>
        </li>
        <li>
          <a
            aria-label="Page 5"
            role="button"
            tabindex="0"
          >
            5
          </a>
        </li>
        <li
          class="next"
        >
          <a
            aria-disabled="false"
            aria-label="Next page"
            rel="next"
            role="button"
            tabindex="0"
          >
            Pr贸ximo
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pagination component should render without data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
